src/core/Checkbox/Checkbox.tsx

```diff
@@ -86,7 +86,7 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
           ? false
           : inputElementRef.current.checked;
       }
-    }, [indeterminate]);
+    });
```

Comment:
Without this change:
![checkbox-with-intermediate](https://user-images.githubusercontent.com/36186912/124287248-76b35000-db58-11eb-98d4-158de38bce29.gif)

With change:
![checkbox-with-intermediate2](https://user-images.githubusercontent.com/36186912/124287268-7c109a80-db58-11eb-9eb1-58e12d62f8d1.gif)

Reply:
Should it really be `indeterminate` though? I would think that it should be `checked` when all selectable rows are selected. ("selectable rows" = non-disabled rows)

Maybe @FlyersPh9 can comment here.

Reply:
that is interesting :O
I would agree, that clicking the all checkbox should go to checked state. Staying indeterminate does not look ok.

Reply:
I have no idea what is the proper way here as antd shows selected state, material-ui indeterminate. Also, all possible rows are selected so it should be checked but on the other hand not all rows are selected, so idk 😃

Reply:
I like that clicking on the Check All checkbox selects first, then deselects on the second click.

The indeterminate VS regular checkmark when a disabled row is displayed… it does seem weird to click & not get the regular checkmark because there is a disabled row present.  I think often times the disabled row may only be temporarily disabled as the file is uploading & processing.  Once that row becomes enabled, do we auto select it?  Or would it remain unselected? 🤔  I would think it would remain unselected, and if that is the case, then the indeterminate checkbox would make sense.

I would like to check how the other big players are handling this.

Reply:
> I like that clicking on the Check All checkbox selects first, then deselects on the second click.

That part is just standard behavior (try on a normal table without any disabled rows).

> I would like to check how the other big players are handling this.

As Vilius said, Material-UI and AntD (the two most popular React UI libraries) have opposite behavior for both `indeterminate` state as well as the "select first, then deselect" part.
https://material-ui.com/components/data-grid/selection/#disable-selection-on-certain-rows
https://ant.design/components/table/#components-table-demo-row-selection

[BlueprintJS](https://blueprintjs.com/docs/#table/api.table), [Semantic UI](https://react.semantic-ui.com/collections/table/#variations-selectable-row), [React-Bootstrap](https://react-bootstrap.github.io/components/table/) and [Fluent UI](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist) don't support this use case.

Reply:
Wouldn't that confuse users when checked becomes indeterminate without user interaction?

Reply:
No.. I dont get it.
Its either checked or none if user clicks on select all checkbox... Disabled is disabled and it cannot be a part of calculations for selected all.

Reply:
Let's go with this approach:
> > Once that row becomes enabled, do we auto select it? Or would it remain unselected?
> 
> it should remain unselected, and the header cell should become `indeterminate` if it was previously `checked`.

Like Vilius said, it may confuse users, but I think it's a good logical starting point and we can test with users later on and change if needed.

Reply:
Changed to show checked when selected all non-disabled rows.